httpserver/main.cpp

```diff
@@ -100,20 +107,43 @@ int main(int argc, char* argv[]) {
             BOOST_LOG_TRIVIAL(info)
                 << "Setting database location to " << dbLocation;
         }
-
+        using endpoint_t = boost::asio::ip::tcp::endpoint;
+        endpoint_t ep;
+        {
+          using err_t = boost::system::error_code;
+          using resolver_t = boost::asio::ip::tcp::resolver;
+          resolver_t r{ioc};
+          err_t ec;
+          auto res = r.resolve(resolver_t::query(argv[1], argv[2]), ec);
+          if(ec)
+          {
+            BOOST_LOG_TRIVIAL(fatal) << "failed to resolve " << argv[1] << ":" << argv[2] << ", " << ec.message() ;
+            return EXIT_FAILURE;
+          }
+          ep = *res;
+        }
         BOOST_LOG_TRIVIAL(info) << "Listening at address " << argv[1]
-                                << " port " << argv[2] << std::endl;
-
-        boost::asio::io_context ioc{1};
+                                << " port " << argv[2] ;
 
         Storage storage(dbLocation);
         ChannelEncryption<std::string> channelEncryption(lokinetIdentityPath);
 
-        tcp::acceptor acceptor{ioc, {address, port}};
+        tcp::acceptor acceptor{ioc, ep};
         tcp::socket socket{ioc};
         http_server(acceptor, socket, storage, channelEncryption);
 
+        closers.emplace_back([&]() {
+            BOOST_LOG_TRIVIAL(info) << "closing..." ;
+            acceptor.close();
+            ioc.stop();
+            BOOST_LOG_TRIVIAL(info) << "bye" ;
+        });
+        
+        signal(SIGINT, handle_signal);
```

Comment:
Is this going to work/compile on Windows too?

Reply:
Just double checking, I wasn't sure if Windows needed different signal handling.